internal/core/hosts/discovered_host.go

```diff
@@ -7,7 +7,7 @@ type DiscoveredHost struct {
 	HostName                 string            `json:"hostname"`
 	CPUCount                 int               `json:"cpu_count"`
 	SocketCount              int               `json:"socket_count"`
-	TotalMemoryMB            int               `json:"total_memory_mb"`
+	TotalMemoryMB            uint64            `json:"total_memory_mb"`
```

Comment:
This change is needed to comply with the new signature of the function `getTotalMemoryMB` (see comment below).

The change is made according to the following rationales:
1. We only support 64-bit architectures, thus `int` is always the same as `int64`
2. `int`, `int64` and `uint64` have the same JSON representation 
3. `uint64` has the same precision loss as `int` when marshalling to JSON, as  their value cannot exceed 2⁵³−1